plugins/wasmedge_zlib/zlibfunc.cpp

```diff
@@ -0,0 +1,1411 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2019-2022 Second State INC
+
+#include "zlibfunc.h"
+
+#include <cstring>
+#include <iostream>
+
+namespace WasmEdge {
+namespace Host {
+
+constexpr bool CheckSize(int32_t StreamSize) {
+
+  return (StreamSize == static_cast<int32_t>(sizeof(WasmZStream)));
+}
+
+static constexpr uint32_t WasmGZFileStart = sizeof(gzFile);
+
+template <typename T>
+auto SyncRun(WasmEdgeZlibEnvironment &Env, uint32_t ZStreamPtr,
+             const Runtime::CallingFrame &Frame, T Callback)
+    -> Expect<int32_t> {
+
+  auto *MemInst = Frame.getMemoryByIndex(0);
+  if (MemInst == nullptr) {
+    return Unexpect(ErrCode::Value::HostFuncError);
+  }
+  WasmZStream *ModuleZStream = MemInst->getPointer<WasmZStream *>(ZStreamPtr);
+
+  const auto HostZStreamIt = Env.ZStreamMap.find(ZStreamPtr);
+  if (HostZStreamIt == Env.ZStreamMap.end()) {
```

Comment:
Use `spdlog` to print the error message.

Reply:
Done.